### DISCUSSION TITLE
What is the purpose of the pixels dictionary member in the Outline examples?

### DISCUSSION THREAD

**Discussion by kallistisoft** [Moddable-OpenSource/moddable#1278] [Getting started] [answered]
I've looked everywhere in the documentation and can't find any description of the `pixels` dictionary member which is passed to the Application class in the `piu/outline` examples.

```js
export default new ClockApplication({}, { pixels: 240 * 32 });
``` 

So... I have several questions about the Application dictionary object:
* What is the purpose of the `pixels` member?
* What is the default value for `pixels`?
* What is the default value for `commandListLength`?
* What is the default value for `displayListLength`?

**Comment by phoddie** [comment, marked as answer]
In brief:

- The `pixels` property defines the number of pixels in the render buffer using by Poco. The [default is 2 scan lines of pixels](https://github.com/Moddable-OpenSource/moddable/blob/9b9e6a9b478bd11e58fe3d2b3d8b987d8ade9925/modules/commodetto/commodettoPocoCore.js#L39). More pixels improves rendering performance. Practically speaking, increasing it much beyond 32 scan lines generally doesn't make much difference however. The details depend on what your project draws, so some experimentation and watching the rendering frame rate is usually appropriate.
- The `commandListLength` is the space reserved (in bytes) for Piu drawing operations. These are high level commands. The [default is 1024](https://github.com/Moddable-OpenSource/moddable/blob/9b9e6a9b478bd11e58fe3d2b3d8b987d8ade9925/modules/piu/MC/piuView.c#L1312-L1313).
- The `displayListLength` is the space reserved (in bytes) for Poco display list operations. These are low level commands. One Piu drawing operation may result in several display list operations, so the display list is sometimes  larger than the command list.  The [default is 1024](https://github.com/Moddable-OpenSource/moddable/blob/9b9e6a9b478bd11e58fe3d2b3d8b987d8ade9925/modules/commodetto/commodettoPocoCore.js#L48).
